kaos/algorithm/view_cone.py

```diff
@@ -1,164 +1,161 @@
 
     # Adjusting the intervals to fit inside the input POI and return
-    return _trim_poi_segments(interval_list, poi)
+    return time_intervals.trim_poi_segments(interval_list, poi)
 
 
-def _trim_poi_segments(interval_list, poi):
-    """Semi-private: Adjusts list of intervals so that all intervals fit inside the poi
+def earth_radius_at_geocetric_lat(geoc_lat):
+    """Calculates Earth's radius at a particular geocentric latitude
 
-        Args:
-            interval_list (list of TimeIntervals): the intervals to be trimmed
-            poi (TimeInterval): period of interest, reference for trimming
+    Args:
+        goc_lat (float): geocentric latitude (Rad)
 
-        Returns:
-            List of TimeIntervals that fit inside the poi
+    Returns:
+        Earth's radius at the given geocentric latitude (in meters)
     """
-    ret_list = []
-    for interval in interval_list:
-        if (interval.start > poi.end) or (interval.end < poi.start):
-            # Outside the input POI
-            continue
-        elif (interval.start < poi.end) and (interval.end > poi.end):
-            ret_list.append(TimeInterval(interval.start, poi.end))
-        elif (interval.end > poi.start) and (interval.start < poi.start):
-            ret_list.append(TimeInterval(poi.start, interval.end))
-        else:
-            ret_list.append(TimeInterval(interval.start, interval.end))
-
-    return ret_list
-
-
-def _view_cone_calc(site_eci, sat_pos, sat_vel, q_magnitude, m):
-    """Semi-private: Performs the viewing cone visibility calculation for the day defined by m.
+    return EARTH_A_AXIS * EARTH_B_AXIS / mp.sqrt(EARTH_A_AXIS ** 2 * mp.sin(geoc_lat) ** 2 +
+                                                 EARTH_B_AXIS ** 2 * mp.cos(geoc_lat) ** 2)
+
 
-    This function is based on a paper titled "rapid satellite-to-site visibility determination
+def _view_cone_calc(lat_geoc, lon_geoc, sat_pos, sat_vel, q_max, m):
+    """Semi-private: Performs the viewing cone visibility calculation for the day defined by m.
+    Note: This function is based on a paper titled "rapid satellite-to-site visibility determination
     based on self-adaptive interpolation technique"  with some variation to account for interaction
     of viewing cone with the satellite orbit.
 
     Args:
-        site_eci (Vector3D): site location in ECI at the start of POI
-        sat_pos (Vector3D): position of satellite (at the same time as sat_vel)
-        sat_vel (Vector3D): velocity of satellite (at the same time as sat_pos)
-        q_magnitude (int): maximum orbital radius
+        lat_geoc(mpf): site location in degrees at the start of POI
+        lon_geoc(mpf): site location in degrees at the start of POI
+        sat_pos(Vector3D): position of satellite (at the same time as sat_vel)
+        sat_vel(Vector3D): velocity of satellite (at the same time as sat_pos)
+        q_max(int): maximum orbital radius
```

Comment:
m(int): time interval offset in days (private, needed for trigonometric calculations)